### ISSUE TITLE
Redlock does not implement / respect namespace

### ISSUE THREAD

**Issue by smlbiobot** [aio-libs/aiocache#520]
I am using trying to use the `aiocache.lock.RedLock` which comes with your library, but it has issues with a namespace is set. As namespace is get/set on the `cache` itself, this lock has issue fetching the key when the namespace is set. (The actual key has a different value depending on whether a namespace is set.

For example, by inserting namespace in your example:

```py
from aiocache import Cache
from aiocache.lock import RedLock

namespace = "foo"

cache = Cache(Cache.REDIS)
async with RedLock(cache, 'key', lease=1):  # Calls will wait here
    result = await cache.get('key', namespace=namespace)
    if result is not None:
        return result
    result = await super_expensive_function()
    await cache.set('key', result, namespace=namespace)
```

It will run into errors when namespace is not None. I have tried to manually insert the namespace into the key used in lock but then the two keys don’t match and the lock throws an error. The only way I see that you can get around this is by allowing `namespace` to be be a kwargs for `RedLock`.

Alternatively, how do you envision that locks are implemented using your library?

Thanks!

**Comment by Dreamsorcerer** [comment]
If you can create a PR with a test and a fix, then we can get this resolved. Adding a namespace parameter sounds reasonable.

**Comment by pshafer-als** [comment]
This should now be fixed by #627.
